### PR TITLE
Gat/mrd pipeline fixes

### PR DESCRIPTION
**Pull request by GatKrieger**
https://ultimagen.atlassian.net/wiki/spaces/IT/pages/2478440711/MRDFeatremap+updates

### CODE REVIEW

File: ugvc/mrd/mrd_utils.py [hunk 1]
```diff
@@ -169,10 +169,18 @@ def collect_coverage_per_locus(coverage_bw_files, df_sig):
     return df_list
 
 
+def collect_coverage_per_locus_gatk(coverage_csv, df_sig):
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py:172]: Please add a docstring

Comment (kenissur) [reply ugvc/mrd/mrd_utils.py:172]: Also, I suggest more general names like "vcf_dataframe" or "signature_dataframe" for clarity

File: ugvc/mrd/mrd_utils.py [hunk 2]
```diff
@@ -169,10 +169,18 @@ def collect_coverage_per_locus(coverage_bw_files, df_sig):
...
+    coverage_gatk = pd.read_csv(coverage_csv, usecols=["Chrom", "Pos", "Total_Depth"])
+    coverage_gatk.rename(columns={"Chrom": "chrom", "Pos": "pos", "Total_Depth": "coverage"}, inplace=True)
+    coverage_gatk.set_index(["chrom", "pos"], inplace=True)
+    df_sig = df_sig.join(coverage_gatk, how="left")
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py:184]: Please make sure the index of df_sig is indeed ["chrom", "pos"], if not raise an exception

File: ugvc/mrd/mrd_utils.py [hunk 3]
```diff
@@ -847,6 +852,8 @@ def prepare_data_from_mrd_pipeline(
         File name or a list of file names, signature vcf files of db (synthetic) control signature/s
     coverage_bw_files: list[str]
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: Remove the coverage_bw_files section

File: ugvc/mrd/mrd_utils.py [hunk 4]
```diff
@@ -974,9 +981,10 @@ def generate_synthetic_signatures(
         trinuc_dict[trinucsub] = {}
         trinuc_dict[trinucsub]["index_to_sample"] = {}
         trinuc_dict[trinucsub]["trinuc_counter"] = {}
+        random_choice_replace = bool(n_subs_db < n_subs_signature)
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: Nice. Where did you come across a signature larger than the db, btw?

You don't need the bool(), this does the same
random_choice_replace = n_subs_db < n_subs_signature

File: ugvc/pipelines/mrd/featuremap_to_dataframe.py [hunk 5]
```diff
@@ -43,52 +43,29 @@ def __parse_args(argv: list[str]) -> argparse.Namespace:
...
         type=str,
-        nargs="+",
         required=False,
-        default="all",
-        help="""List of input info fields to include in dataframe, by default 'all''
-        If 'all' then all the info fields are read to columns
-        If None then no info fields are read to columns""",
+        default=None,
+        help="""Comma-separated list of input info fields to include in dataframe,
+        If None (default) then all the info fields are read to columns""",
```
Comment (kenissur) [inline ugvc/pipelines/mrd/featuremap_to_dataframe.py:49]: Why this change? default=None is good, but no reason for the comma separated list, with nargs='+' you can do:
--info_fields field1 field2 field3 
etc.

File: ugvc/pipelines/mrd/featuremap_to_dataframe.py [hunk 6]
```diff
@@ -43,52 +43,29 @@ def __parse_args(argv: list[str]) -> argparse.Namespace:
...
     )
 
     parser.add_argument(
-        "--input-format-fields", type=str, nargs="+", default=None, help="Fields to extract from the vcf FORMAT fields"
+        "--input-format-fields",
+        type=str,
+        default=None,
+        help="comma-separated list of fields to extract from the vcf FORMAT fields, by default none are read",
```
Comment (kenissur) [inline ugvc/pipelines/mrd/featuremap_to_dataframe.py]: Here too, please revert this change

File: ugvc/pipelines/mrd/prepare_data_from_mrd_pipeline.py [hunk 7]
```diff
@@ -56,13 +56,11 @@ def parse_args(argv: list[str]) -> argparse.Namespace:
...
     parser.add_argument(
-        "-c",
-        "--coverage-bw",
+        "--coverage-csv",
         type=str,
-        nargs="+",
         default=None,
         required=False,
-        help="Coverage bigwig files generated with 'coverage_analysis full_analysis'",
+        help="Coverage csv file generated with ExtractCoverageOverVcfFiles",
```
Comment (kenissur) [inline ugvc/pipelines/mrd/prepare_data_from_mrd_pipeline.py]:
```suggestion
        help="Coverage csv file generated with gatk ExtractCoverageOverVcfFiles",
```

**Review by kenissur — APPROVED** [review]
Looks good, just a few minor comments, please address before merging. Also, please make sure the WDLs run with the most recent docker here before merging.

File: ugvc/mrd/mrd_utils.py (hunk 4 above)
Comment (GatKrieger) [reply ugvc/mrd/mrd_utils.py]: In one of the Tempus patients, I think...don't fully remember.
